Crash Report - azapi_v2.9.0 - Landing Zone Deployment

Just reporting this as requested - happened during a reasonably large update to a Landing Zone

```

Plan: 104 to add, 0 to change, 0 to destroy.
╷
│ Error: Plugin did not respond
│ 
│ The plugin encountered an error, and failed to respond to the plugin6.(*GRPCProvider).ReadResource call. The plugin logs may contain more details.
╵
╷
│ Error: Plugin did not respond
│ 
│ The plugin encountered an error, and failed to respond to the plugin6.(*GRPCProvider).ReadResource call. The plugin logs may contain more details.
╵
╷
│ Error: Plugin did not respond
│ 
│ The plugin encountered an error, and failed to respond to the plugin6.(*GRPCProvider).ReadResource call. The plugin logs may contain more details.
╵
╷
│ Error: Plugin did not respond
│ 
│ The plugin encountered an error, and failed to respond to the plugin6.(*GRPCProvider).PlanResourceChange call. The plugin logs may contain more details.
╵
Releasing state lock. This may take a few moments...

Stack trace from the terraform-provider-azapi_v2.9.0 plugin:

panic: runtime error: index out of range [0] with length 0

goroutine 241 [running]:
github.com/Azure/terraform-provider-azapi/utils.updateObjectAtPath({0x114c1e2c0, 0x14001a0e0a8}, {0x114c1e2c0, 0x14000331da0}, {0x0, 0x1, 0x0, 0x0, 0x0}, {0x14000d3b0b0, ...})
        github.com/Azure/terraform-provider-azapi/utils/json.go:159 +0xeb0
github.com/Azure/terraform-provider-azapi/utils.updateObjectAtPath({0x114cb0b20, 0x140011244b0}, {0x114cb0b20, 0x14001124330}, {0x0, 0x1, 0x0, 0x0, 0x0}, {0x140022a0891, ...})
        github.com/Azure/terraform-provider-azapi/utils/json.go:145 +0x834
github.com/Azure/terraform-provider-azapi/utils.updateObjectAtPath({0x114cb0b20, 0x14001124360}, {0x114cb0b20, 0x14001124300}, {0x0, 0x1, 0x0, 0x0, 0x0}, {0x0, ...})
        github.com/Azure/terraform-provider-azapi/utils/json.go:145 +0x834
github.com/Azure/terraform-provider-azapi/utils.UpdateObject(...)
        github.com/Azure/terraform-provider-azapi/utils/json.go:128
github.com/Azure/terraform-provider-azapi/internal/services.(*AzapiResource).Read(0x14001688b68, {0x114e092a8, 0x14000ca11a0}, {{{{0x114e0f378, 0x14000a4d770}, {0x114cb1cc0, 0x14000a4c480}}, {0x114e11a88, 0x14000180c30}}, 0x140009e2ba0, ...}, ...)
        github.com/Azure/terraform-provider-azapi/internal/services/azapi_resource.go:1112 +0x1634
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).ReadResource(0x14000264308, {0x114e092a8, 0x14000ca11a0}, 0x140019c91d0, 0x14000b255d0)
        github.com/hashicorp/terraform-plugin-framework@v1.16.1/internal/fwserver/server_readresource.go:156 +0x918
github.com/hashicorp/terraform-plugin-framework/internal/proto6server.(*Server).ReadResource(0x14000264308, {0x114e092a8?, 0x14000ca10b0?}, 0x1400179f180)
        github.com/hashicorp/terraform-plugin-framework@v1.16.1/internal/proto6server/server_readresource.go:71 +0x434
github.com/hashicorp/terraform-plugin-go/tfprotov6/tf6server.(*server).ReadResource(0x14000262500, {0x114e092a8?, 0x14000e5f170?}, 0x14000934070)
        github.com/hashicorp/terraform-plugin-go@v0.29.0/tfprotov6/tf6server/server.go:863 +0x210
github.com/hashicorp/terraform-plugin-go/tfprotov6/internal/tfplugin6._Provider_ReadResource_Handler({0x114ddd1e0, 0x14000262500}, {0x114e092a8, 0x14000e5f170}, 0x14000ecc980, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.29.0/tfprotov6/internal/tfplugin6/tfplugin6_grpc.pb.go:753 +0x1c0
google.golang.org/grpc.(*Server).processUnaryRPC(0x1400023e400, {0x114e092a8, 0x14000e5f0e0}, 0x14001256420, 0x1400032d740, 0x11551e838, 0x0)
        google.golang.org/grpc@v1.75.1/server.go:1431 +0xc9c
google.golang.org/grpc.(*Server).handleStream(0x1400023e400, {0x114e0a158, 0x1400047c000}, 0x14001256420)
        google.golang.org/grpc@v1.75.1/server.go:1842 +0x900
google.golang.org/grpc.(*Server).serveStreams.func2.1()
        google.golang.org/grpc@v1.75.1/server.go:1061 +0x84
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 39
        google.golang.org/grpc@v1.75.1/server.go:1072 +0x138

Error: The terraform-provider-azapi_v2.9.0 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.`
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -8,6 +8,10 @@ ENHANCEMENTS:
 - Bump `terraform-plugin-framework` from v1.16.1 to v1.19.0 (GH-1081).
 - Add example for `Microsoft.RedHatOpenShift/openShiftClusters` (GH-1083).
 
+BUG FIXES:
+
+- Fix a panic when comparing a non-empty configured list with an empty remote list during no-op change detection (GH-1125).
+
 ## v2.9.0
 
 ENHANCEMENTS:

--- a/utils/json.go
+++ b/utils/json.go
@@ -151,8 +151,8 @@ func updateObjectAtPath(old interface{}, new interface{}, option UpdateJsonOptio
 		}
 	case []interface{}:
 		if newArr, ok := new.([]interface{}); ok {
-			if len(oldValue) == 0 {
-				return new
+			if len(oldValue) == 0 || len(newArr) == 0 {
+				return newArr
 			}
 
 			identifierKey := listIdentifierKeyForPath(path, option)

--- a/utils/json_test.go
+++ b/utils/json_test.go
@@ -1075,6 +1075,49 @@ func Test_UpdateObjectDuplicateIdentifiersWithInconsistentOrdering(t *testing.T)
 	}
 }
 
+func Test_UpdateObjectNonEmptyOldArrayWithEmptyNewArray(t *testing.T) {
+	OldJson := `
+{
+  "properties": {
+    "subnets": [
+      {
+        "name": "default",
+        "properties": {
+          "addressPrefix": "10.0.3.0/24"
+        }
+      }
+    ]
+  }
+}
+`
+	NewJson := `
+{
+  "properties": {
+    "subnets": []
+  }
+}
+`
+	ExpectJson := `
+{
+  "properties": {
+    "subnets": []
+  }
+}
+`
+
+	var old, new, expected any
+	_ = json.Unmarshal([]byte(OldJson), &old)
+	_ = json.Unmarshal([]byte(NewJson), &new)
+	_ = json.Unmarshal([]byte(ExpectJson), &expected)
+
+	got := utils.UpdateObject(old, new, utils.UpdateJsonOption{})
+	if !reflect.DeepEqual(got, expected) {
+		expectedJson, _ := json.MarshalIndent(expected, "", "  ")
+		gotJson, _ := json.MarshalIndent(got, "", "  ")
+		t.Fatalf("Expected:\n%s\n\n but got\n%s", expectedJson, gotJson)
+	}
+}
+
 func Test_UpdateObject_ListUniqueIdProperty_IgnoreOtherItemsInList(t *testing.T) {
 	testcases := []struct {
 		Name                   string

